### PR TITLE
VACMS-18336: Remove flipper for typeahead search

### PR DESCRIPTION
**Pull request by chriskim2311**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
Remove the `search_dropdown_component_enabled` flipper and fix logic to not use it anymore.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18336

## What areas of the site does it impact?
Search

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

**Review by randimays — COMMENTED** [review]
Nice cleanup! Just a couple of comments.

**Review by powellkerry — DISMISSED** [review]

File: src/platform/site-wide/header/components/Search/index.js
Comment (chriskim2311) [reply src/platform/site-wide/header/components/Search/index.js:76]: @laflannery Can you take a look at this comment from Randi thanks!

Comment (laflannery) [reply src/platform/site-wide/header/components/Search/index.js:76]: @chriskim2311 I feel like I don't understand this unless I look at it in real life - will the review instance show me this new code and what was removed?

Comment (laflannery) [reply src/platform/site-wide/header/components/Search/index.js:76]: Ah I see! The visible label is actually not being used by AT and instead we have an `aria-label`:
image
You can actually remove the `aria-label` here (mobile only, we do need it on desktop) and then change the ID as @randimays suggested

**Review by laflannery — APPROVED** [review]
Looks good

**Review by randimays — APPROVED** [review]
LGTM!

**Comment by randimays** [comment]
@chriskim2311 Is this PR ready to merge? I have another PR for search coming in behind it; wanting to make sure I properly pick up your changes.

**Comment by randimays** [comment]
Clarifying the above question - is it ready to merge after Platform approval?*

**Comment by chriskim2311** [comment]
@randimays Yep! No more changes you can pull the branch in if you like. I'm opening a plat ticket today.

**Review by harshilmoradia — APPROVED** [review]

**Pull request merged**
